### PR TITLE
feat(compute): Samples for start stop instances documentation page

### PR DESCRIPTION
**Pull request by rsamborski**
New start/stop samples for: https://cloud.google.com/compute/docs/instances/stop-start-instance

### CODE REVIEW

File: compute/cloud-client/instances/src/create_instance_with_encryption_key.php [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+
+/**
+ * For instructions on how to run the full sample:
+ *
+ * @see https://github.com/GoogleCloudPlatform/php-docs-samples/tree/master/compute/cloud-client/README.md
+ */
+
+namespace Google\Cloud\Samples\Compute;
+
+# [START compute_instances_create_with_encryption_key]
```
Comment (savijatv) [inline compute/cloud-client/instances/src/create_instance_with_encryption_key.php:26]: We've another region tag (_compute_instances_create_encrypted_) for encrypted instance (Java code is available in DevRel snippet tracker - https://devrel.corp.google.com/snippets/compute_instances_create_encrypted?product_id=48). Is this PHP sample different from that Java sample? If not, we might want to use one tag for all lang samples.

File: compute/cloud-client/instances/src/create_instance_with_encryption_key.php [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+use Google\Cloud\Compute\V1\NetworkInterface;
+
+/**
+ * To correctly handle string enums in Cloud Compute library
+ * use constants defined in the Enums subfolder.
+ */
+use Google\Cloud\Compute\V1\Enums\AttachedDisk\Type;
+
+/**
+ * Creates an instance in the specified project and zone with boot disk encrypted with customer provided key.
```
Comment (savijatv) [inline compute/cloud-client/instances/src/create_instance_with_encryption_key.php:41]: Can we rephrase this as 'Creates an instance with encrypted disk in the ....'

File: compute/cloud-client/instances/src/create_instance_with_encryption_key.php [hunk 3]
```diff
@@ -0,0 +1,115 @@
+<?php
+/**
+ * Copyright 2021 Google Inc.
```
[the remark this answers is not in the document]
Comment (rsamborski) [reply compute/cloud-client/instances/src/create_instance_with_encryption_key.php:3]: Good catch.

File: compute/cloud-client/instances/src/create_instance_with_encryption_key.php (hunk 2 above)
Comment (rsamborski) [reply compute/cloud-client/instances/src/create_instance_with_encryption_key.php:41]: I have rephrased to 'Creates an instance with encrypted disk customer provided key in the specified project and zone.'.

I want to emphasise the customer provided key as all disks in GCP are encrypted, but normally they use Google provided keys.

Comment (rsamborski) [reply compute/cloud-client/instances/src/create_instance_with_encryption_key.php:41]: PLMK if it's ok

File: compute/cloud-client/instances/src/start_instance_with_encryption_key.php [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+    $diskData = (new CustomerEncryptionKeyProtectedDisk())
+        ->setSource($instanceData->getDisks()[0]->getSource())
+        ->setDiskEncryptionKey($customerEncryptionKey);
+
+    // Set request with one disk.
+    $instancesStartWithEncryptionKeyRequest = (new InstancesStartWithEncryptionKeyRequest())
+        ->setDisks(array($diskData));
+
+    // Start the instance with encrypted disk.
+    $operation = $instancesClient->startWithEncryptionKey($instanceName, $instancesStartWithEncryptionKeyRequest, $projectId, $zone);
```
[the remark this answers is not in the document]
Comment (rsamborski) [reply compute/cloud-client/instances/src/start_instance_with_encryption_key.php:77]: See above

File: compute/cloud-client/instances/src/reset_instance.php [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+ * @throws \Google\ApiCore\ValidationException if local error occurs before remote call.
+ */
+function reset_instance(
+    string $projectId,
+    string $zone,
+    string $instanceName
+) {
+    // Stop the Compute Engine instance using InstancesClient.
+    $instancesClient = new InstancesClient();
+    $operation = $instancesClient->reset($instanceName, $projectId, $zone);
```
[the remark this answers is not in the document]
Comment (rsamborski) [reply compute/cloud-client/instances/src/reset_instance.php:51]: See above

File: compute/cloud-client/instances/src/create_instance_with_encryption_key.php (hunk 2 above)
Comment (savijatv) [reply compute/cloud-client/instances/src/create_instance_with_encryption_key.php]: Yes, this looks fine. Just a nit.  I think a conjunction is missing.
Creates an instance with encrypted disk **that uses** (?) a customer provided key in the specified ....

File: compute/cloud-client/instances/src/start_instance_with_encryption_key.php [hunk 6]
```diff
@@ -0,0 +1,91 @@
...
+ *
+ * Example:
+ * ```
+ * start_instance_with_encryption_key($projectId, $zone, $instanceName, $key);
+ * ```
+ *
+ * @param string $projectId Project ID or project number of the Cloud project your instance belongs to.
+ * @param string $zone Name of the zone your instance belongs to.
+ * @param string $instanceName Name of the instance your want to stop.
+ * @param string $key Bytes object representing a raw base64 encoded key to your machines boot disk.
```
[the remark this answers is not in the document]
Comment (savijatv) [reply compute/cloud-client/instances/src/start_instance_with_encryption_key.php:43]: Apostrophe still missing? .. machine's boot disk.

File: compute/cloud-client/instances/src/start_instance_with_encryption_key.php [hunk 7]
```diff
@@ -0,0 +1,91 @@
...
+ * Starts a stopped Google Compute Engine instance (with encrypted disks).
+ *
+ * Example:
+ * ```
+ * start_instance_with_encryption_key($projectId, $zone, $instanceName, $key);
+ * ```
+ *
+ * @param string $projectId Project ID or project number of the Cloud project your instance belongs to.
+ * @param string $zone Name of the zone your instance belongs to.
+ * @param string $instanceName Name of the instance your want to stop.
```
Comment (savijatv) [inline compute/cloud-client/instances/src/start_instance_with_encryption_key.php:42]: .. instance you want to stop.

File: compute/cloud-client/instances/src/start_instance_with_encryption_key.php (hunk 6 above)
[the remark this answers is not in the document]
Comment (rsamborski) [reply compute/cloud-client/instances/src/start_instance_with_encryption_key.php]: Fixed

File: compute/cloud-client/instances/src/create_instance_with_encryption_key.php (hunk 2 above)
Comment (rsamborski) [reply compute/cloud-client/instances/src/create_instance_with_encryption_key.php]: Fixed

File: compute/cloud-client/instances/src/start_instance_with_encryption_key.php (hunk 7 above)
Comment (rsamborski) [reply compute/cloud-client/instances/src/start_instance_with_encryption_key.php]: Fixed

**Review by savijatv — APPROVED** [review]

File: compute/cloud-client/instances/src/start_instance.php [hunk 8]
```diff
@@ -0,0 +1,65 @@
...
+# [START compute_start_instance]
+use Google\Cloud\Compute\V1\InstancesClient;
+
+/**
+ * Starts a stopped Google Compute Engine instance (with unencrypted disks).
+ *
+ * Example:
+ * ```
+ * start_instance($projectId, $zone, $instanceName);
+ * ```
```
Comment (bshaffer) [inline compute/cloud-client/instances/src/start_instance.php:35]: I really don't think these 'examples' add any value here, since we have the actual example below (which is really the point), and the parameters are a part of the doc block. I would prefer to remove all of these.

**Review by bshaffer — COMMENTED** [review]
One suggestion, otherwise REALLY great job on these!

File: compute/cloud-client/instances/src/start_instance.php (hunk 8 above)
Comment (rsamborski) [reply compute/cloud-client/instances/src/start_instance.php]: I agree that these examples don't bring any value plus they are painfully tedious in keeping in sync with function parameters. I have removed them from all Compute samples.

**Review by bshaffer — APPROVED** [review]
